aws/static_site/main.tf

```diff
@@ -20,11 +20,16 @@ resource "aws_s3_bucket" "mod" {
     }
   )
 
-  cors_rule {
-    allowed_headers = ["*"]
-    allowed_methods = ["GET"]
-    allowed_origins = var.enable_cors_get ? ["*"] : []
-    max_age_seconds = 3000
+  dynamic "cors_rule" {
+    for_each = var.enable_cors_get ? var.cors_rule : []
```

Comment:
Since we’re introducing a new variable (`cors_rule`) here, can we get rid of `enable_cors_get` altogether and base this conditional on whether `cors_rule` has entries or not, instead?

Looking at the variables and the default for `cors_rule` being allow-all-gets-from-everywhere would I think make it easy to miss that that’s not the actual default because there is another variable for opting-in.

Reply:
Principal goal was the make this backwards compatible. Changing the name would not do that. I think we should rename / change / remove the variable (nb: i don't think we should have cors_rule as the primary interface. Named variables/configuration that describe the intent of the cors configuration are more approachable) in some later effort.

Reply:
I am not quite sure I follow that (I see `enable_cors_get` used only in one place, and I think that all other uses of `//aws/static_site` would be unaffected), but if you are thinking about changing this anyway, okay! 😄